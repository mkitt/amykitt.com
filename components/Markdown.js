// @flow
import React from 'react'
import Markdown from 'react-markdown'
import css, { media3, select } from '../styles/css'

type Props = {
  source: string,
}

// Markdown styles
const style = css(
  {
    fontSize: 14,
    position: 'relative',
    color: '#858585',
  },
  select('& h1, & h2', {
    marginTop: 50,
    lineHeight: 1,
    fontSize: 15,
    fontWeight: 700,
    textTransform: 'uppercase',
  }),
  select('& h3', {
    fontSize: 15,
    fontWeight: 400,
    lineHeight: 1,
    textTransform: 'uppercase',
  }),
  select('& img', {
    maxWidth: '100%',
    height: 'auto',
    fontSize: 12,
    marginBottom: 2,
  }),
  select('& ul', {
    margin: 0,
    padding: 0,
    listStyle: 'none',
  }),
  select('& img[alt="Amy Kitt"]', {
    width: 120,
    height: 120,
    fontSize: 12,
  }),
  media3(select('& img[alt="Amy Kitt"]', {
    position: 'absolute',
    top: 0,
    left: 0,
    width: 240,
    height: 720,
  })),
  media3(select('#about &', {
    paddingLeft: 260,
  })),
  select('#about & h1', {
    fontSize: 18,
  }),
)

export default ({ source, ...props }: Props) => (
  <Markdown
    className={`${style}`}
    source={source}
    {...props}
  />
)
