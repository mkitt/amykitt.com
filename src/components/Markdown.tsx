import styled from '@emotion/styled/macro'
import { memo } from 'react'
import ReactMarkdown from 'react-markdown'
import Heading from './Heading'
import Img from './Img'
import View from './View'

type HeadingsPropsType = {
  readonly level: number
}

const Root = styled(View)({
  color: '#858585',
  fontSize: '0.875rem',
  position: 'relative',
  '@media (min-width: 47.5em)': {
    '#about &': {
      paddingLeft: 260,
    },
  },
})

const Title = styled(Heading)({
  '#about &': {
    fontSize: '1.125rem',
  },
})

const Subheading = styled(Heading)({
  fontWeight: 400,
})

const StyledImg = styled(Img)({
  marginBottom: '0.25rem',
  '&[alt="Amy Kitt"]': {
    width: 120,
    height: 241,
  },
  '@media (min-width: 47.5em)': {
    '&[alt="Amy Kitt"]': {
      height: 483,
      left: 0,
      position: 'absolute',
      top: 0,
      width: 240,
    },
  },
})

const List = styled('ul')({
  listStyle: 'none',
  margin: 0,
  padding: 0,
})

const Link = styled.a({
  backgroundColor: 'transparent',
  color: 'inherit',
})

const HeadingRenderer = (props: HeadingsPropsType) => {
  const { level } = props
  const headings = [Title, Heading, Subheading]
  const Comp = headings[level - 1] || Heading
  // @ts-ignore
  return <Comp as={`h${level}`} {...props} />
}

const components = {
  heading: (props: any) => <HeadingRenderer {...props} />,
  image: (props: any) => <StyledImg {...props} />,
  link: (props: any) => <Link {...props} />,
  linkReference: (props: any) => <Link {...props} />,
  list: (props: any) => <List {...props} />,
  root: (props: any) => <Root {...props} />,
}

type PropsType = {
  readonly source: string
}

const Markdown = ({ source }: PropsType) => (
  <ReactMarkdown renderers={components} skipHtml source={source} />
)

export default memo(Markdown)
