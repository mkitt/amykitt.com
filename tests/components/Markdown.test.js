import React from 'react'
import renderer from 'react-test-renderer'
import Markdown from '../../components/Markdown'

describe('Markdown', () => {
  let component

  beforeEach(() => {
    component = <Markdown source="Hello _world_." />
  })

  it('shows the correct snapshot for the Markdown component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
