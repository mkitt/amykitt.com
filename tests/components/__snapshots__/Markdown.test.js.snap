// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown shows the correct snapshot for the Markdown component tree 1`] = `
<div
  className="css-jlylsu"
>
  <p>
    Hello 
    <em>
      world
    </em>
    .
  </p>
</div>
`;
